xbmc/ContextMenuManager.cpp

```diff
@@ -208,13 +208,19 @@ bool CContextMenuManager::IsVisible(
   return menuItem.IsVisible(fileItem);
 }
 
-bool CContextMenuManager::HasItems(const CFileItem& fileItem) const
+bool CContextMenuManager::HasItems(const CFileItem& fileItem,
+                                   const CContextMenuItem& root /*= MAIN*/) const
 {
-  std::unique_lock<CCriticalSection> lock(m_criticalSection);
-  return std::any_of(m_items.cbegin(), m_items.cend(),
-                     [&fileItem](const std::shared_ptr<IContextMenuItem>& menu) {
-                       return menu->IsVisible(fileItem);
-                     });
+  //! @todo implement group support
+  if (&root == &MAIN)
```

Comment:
You did not implement this but hese groups should be namespaced

Reply:
You got it all wrong! See the implementation (not done by me) from the respective getters! It contains exactly the same todo. May I suggest that you next time read what I write in the summary, especially to give you the context for the review?

> @enen92 please review, thanks. The Has methods need to do what the corresponding Get methods (which are available for a long time) do, but only shall check with std::any_of, not actually collect all items.